iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -1333,34 +1325,5 @@ internal static string PopulateMessagePropertiesFromMessage(string topicName, Te
                 : "/";
             return $"{topicName}{properties}{suffix}";
         }
-
-        /// <summary>
-        /// Gets the result of the provided task completion source or throws OperationCanceledException if the provided
-        /// cancellation token is cancelled beforehand.
-        /// </summary>
-        /// <typeparam name="T">The type of the result of the task completion source.</typeparam>
-        /// <param name="taskCompletionSource">The task completion source to asynchronously wait for the result of.</param>
-        /// <param name="timeoutErrorMessage">The error message to put in the OperationCanceledException if this taks times out.</param>
-        /// <param name="cancellationToken">The cancellation token.</param>
-        /// <returns>The result of the provided task completion source if it completes before the provided cancellation token is cancelled.</returns>
-        /// <exception cref="OperationCanceledException">If the cancellation token is cancelled before the provided task completion source finishes.</exception>
-        private static async Task<T> GetTaskCompletionSourceResultAsync<T>(
```

Comment:
By replacing this with a client-wide helper, we don't need a specific case for it. However, this behavior included a custom error message, which a normal canceled call would not normally get, so I feel it is okay to lose that behavior.